core/src/engine/task/task_context.cpp

```diff
@@ -319,6 +320,10 @@ TaskContext::WakeupSource TaskContext::Sleep(WaitStrategy& wait_strategy) {
                             (!IsCancellable() || deadline < cancel_deadline_);
   if (has_deadline) ArmDeadlineTimer(deadline, sleep_epoch);
 
+  // We set this here to find a span from sleeping coroutine
+  // when debugging externally.
+  current_span_within_sleep_ = tracing::Span::CurrentSpanUnchecked();
```

Comment:
that's not good to add a dependency engine -> tracing
maybe create the same current span on each span creation/deletion in Span ctr/dtr and set it into task local variable?